src/test/java/com/commercetools/paymenttoorderprocessor/jobs/actions/MessageReaderTest.java

```diff
@@ -102,33 +98,29 @@ public void read_whenOnePage_returnsResultFromThePage() {
 
         // sphere client should be called only twice (first and empty result page)
         // even if we call messageReader.read 3 times, because both results are on the same page first page
-        verify(client, times(2)).executeBlocking(any());
+        verify(client, times(1)).execute(any());
 
         // no messages processed - last timestamp is not updated
         timeStampManager.persistLastProcessedMessageTimeStamp();
         assertThat(timeStampManager.getLastProcessedMessageTimeStamp()).isNull();
     }
 
     @Test
-    public void read_whenFirstPageIsEmpty_returnsResultFromSecondPage_AndPersistsTimestamp() {
+    public void read_whenSecondMessageIsProcessed_returnsFirstMessage_AndPersistsTimestampOfSecondMessage() {
```

Comment:
Why does your implementation completely change this test? Isn't it just updating a way how it paginates?

Reply:
no, now all messages are loaded at once and fill up the queue and then it is returning the result one by one

Reply:
2463500

Reply:
And how was it before?

Reply:
The queue was filled up page-by-page.So when all elements of the first page of the query was popped out of the queue, the next page was queried and so on.

Reply:
Then shouldn't you do the same thing? If you fetch everything, it might be too many messages? WDYT?

Reply:
but the task said that the queryAll-feature should be used

Reply:
Actually the Resolution is `Use pagination by ID` :) queryAll is just a suggestion. I'm not saying it's wrong, just a question.

Reply:
I think the current logic is now very clean and simple and the job should run often, because a paid cart should be transformed to an order  quickly. So I think there wouldn't be much messages to process. So I would leave the code like it is now.